js/repl/src/Repl.js

```diff
@@ -0,0 +1,413 @@
+// @flow
+
+import { css } from 'glamor';
+import React from 'react';
+import CodeMirrorPanel from './CodeMirrorPanel';
+import ReplOptions from './ReplOptions';
+import StorageService from './StorageService';
+import UriUtils from './UriUtils';
+import compile from './compile';
+import loadPlugin from './loadPlugin';
+import {
+  envPresetConfig,
+  pluginConfigs,
+  presetPluginConfigs,
+  runtimePolyfillConfig
+} from './PluginConfig';
+import {
+  envConfigToTargetsString,
+  getDebugInfoFromEnvResult,
+  loadPersistedState,
+  configArrayToStateMap,
+  configToState,
+  persistedStateToEnvConfig
+} from './replUtils';
+import { media } from './styles';
+
+import type {
+  BabelPresets,
+  EnvConfig,
+  PluginState,
+  PluginStateMap,
+  BabelPresetEnvResult
+} from './types';
+
+type Props = {};
+type State = {
+  builtIns: boolean,
+  code: string,
+  compiled: ?string,
+  compileError: ?Error,
+  debugEnvPreset: boolean,
+  envConfig: EnvConfig,
+  envPresetDebugInfo: ?string,
+  envPresetState: PluginState,
+  evalError: ?Error,
+  isSidebarExpanded: boolean,
+  lineWrap: boolean,
+  plugins: PluginStateMap,
+  presets: PluginStateMap,
+  runtimePolyfillState: PluginState
+};
+
+export default class Repl extends React.Component {
+  static defaultProps = {
+    defaultValue: ''
+  };
+
+  props: Props;
+  state: State;
+
+  _numLoadingPlugins = 0;
+
+  constructor(props: Props, context: any) {
+    super(props, context);
+
+    const persistedState = loadPersistedState();
+
+    const defaultPlugins = {
+      'babili-standalone': persistedState.babili,
+      prettier: persistedState.prettier
+    };
+
+    const defaultPresets = (persistedState.presets || 'es2015,react,stage-2')
+      .split(',')
+      .reduce((reduced, key) => {
+        reduced[`babel-preset-${key}`] = true;
+        return reduced;
+      }, {});
+
+    const envConfig = persistedStateToEnvConfig(persistedState);
+
+    const state = {
+      builtIns: persistedState.builtIns,
+      code: persistedState.code,
+      compiled: null,
+      compileError: null,
+      debugEnvPreset: persistedState.debug,
+      envConfig,
+      envPresetDebugInfo: null,
+      envPresetState: configToState(
+        envPresetConfig,
+        envConfig.isEnvPresetEnabled
+      ),
+      evalError: null,
+      isSidebarExpanded: persistedState.showSidebar,
+      lineWrap: persistedState.lineWrap,
+      plugins: configArrayToStateMap(pluginConfigs, defaultPlugins),
+      presets: configArrayToStateMap(presetPluginConfigs, defaultPresets),
+      runtimePolyfillState: configToState(
+        runtimePolyfillConfig,
+        persistedState.evaluate
+      )
+    };
+
+    this.state = {
+      ...state,
+      ...this._compile(persistedState.code, state)
```

Comment:
Not silly 😅 I forgot why I'd done it this way too. So the comment will help future-me.